### PR TITLE
Typing: `rigid_body`

### PR DESCRIPTION
**Pull request by sy-cui**
* Added mypy typehints for `elastica/rigid_body`
* Needs to add typehints for `mesh_rigid_body.py`

### CODE REVIEW

File: elastica/rigidbody/data_structures.py [hunk 1]
```diff
@@ -51,83 +49,83 @@ def __init__(self) -> None:
...
-    omega = np.ndarray.view(vector_states[..., n_velocity_end:n_omega_end])
-
-    n_acceleration_end = n_omega_end + n_nodes
-    acceleration = np.ndarray.view(vector_states[..., n_omega_end:n_acceleration_end])
-
-    n_alpha_end = n_acceleration_end + n_elems
-    alpha = np.ndarray.view(vector_states[..., n_acceleration_end:n_alpha_end])
-
-    return output + (position, directors, velocity, omega, acceleration, alpha)
+# def _bootstrap_from_data(stepper_type: str, n_elems: int, vector_states, matrix_states):
```
Comment (skim0119) [inline elastica/rigidbody/data_structures.py:56]: Just remove all comments below. I feel like `data_structures` needs to move to timestepper, and both `rigid_body` and `rod` can import from same file. I can refactor that later with rod module.

**Review by skim0119 — APPROVED** [review]
Few minore change. Otherwise, looks good

File: elastica/rigidbody/data_structures.py (hunk 1 above)
Comment (sy-cui) [reply elastica/rigidbody/data_structures.py]: [d7da645](https://github.com/GazzolaLab/PyElastica/pull/395/commits/d7da64525e2ceacd6c3c246407bbf4ec42e708e8)

Comment (sy-cui) [reply elastica/rigidbody/data_structures.py]: [d7da645](https://github.com/GazzolaLab/PyElastica/pull/395/commits/d7da64525e2ceacd6c3c246407bbf4ec42e708e8)

**Comment by skim0119** [comment]
I'm merging this. I checked `mypy` and `pytest`

**Pull request merged**